dev-tools/packaging/templates/deb/elastic-agent.sh.tmpl

```diff
@@ -23,6 +23,8 @@ WRAPPER="/usr/share/${NAME}/bin/${NAME}-god"
 BEAT_USER="{{.BeatUser}}"
 WRAPPER_ARGS="-r / -n -p $PIDFILE"
 SCRIPTNAME=/etc/init.d/{{.ServiceName}}
+DEFAULT_GODEBUG="madvdontneed=1"
+export GODEBUG=${GODEBUG-$DEFAULT_GODEBUG}
```

Comment:
Usually, environment overrides are done in `/etc/default` files, so this should be done after reading `/etc/default/{{.ServiceName}}` some lines below.